components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -248,13 +248,10 @@ abstract class AbstractFetchDownloadService : Service() {
 
         // Create a new job and add it, with its downloadState to the map
         val downloadJobState = DownloadJobState(
-            state = download,
+            state = download.copy(status = DOWNLOADING),
             foregroundServiceId = foregroundServiceId
         )
 
-        val newState = downloadJobState.state.copy(status = Status.DOWNLOADING)
-        updateDownloadState(newState)
```

Comment:
@kglazko this is the change that I commented you about on slack. We were trying to use `updateDownloadState` to early as `updateDownloadState` update the state on `downloadJobs` but downloadJobState gets added much after in line 259